xlml/apis/task.py

```diff
@@ -70,14 +70,91 @@ def run(self) -> DAGNode:
         group_id=self.task_test_config.benchmark_id, prefix_group_id=True
     ) as group:
       provision, queued_resource, ssh_keys = self.provision()
-      run_model = self.run_model(queued_resource, ssh_keys)
       post_process = self.post_process()
       clean_up = self.clean_up(queued_resource)
 
-      provision >> run_model >> post_process >> clean_up
+      if self.task_test_config.startup_script:
+        check_if_startup_script_end = self.check_if_startup_script_end(
+            queued_resource, ssh_keys
+        )
+        provision >> check_if_startup_script_end >> post_process >> clean_up
+      else:
+        run_model = self.run_model(queued_resource, ssh_keys)
+        provision >> run_model >> post_process >> clean_up
 
     return group
 
+  def check_if_startup_script_end(
+      self,
+      queued_resource: airflow.XComArg,
+      ssh_keys: airflow.XComArg,
+  ) -> DAGNode:
+    """Run the TPU test in `task_test_config`.
+
+    Args:
+      queued_resource: XCom value for the queued resource name (string).
+      ssh_keys: And XCom value for the TPU's SSH keys (SshKeys).
+
+    Returns:
+      A DAG node that executes the model test.
+    """
+
+    check_script = """
```

Comment:
Please don't do this here. If you really want a bash script in the future, you could put them in this folder: https://github.com/GoogleCloudPlatform/ml-auto-solutions/tree/master/scripts

Reply:
Based on my understanding, everything in https://github.com/GoogleCloudPlatform/ml-auto-solutions/tree/master/scripts are the scripts we run on cloudtop. 

But I am running `check_script` on a TPU VM. 

`check_if_startup_script_end`'s job is to check if the workload from TPU's startup script finish, it might be easier to put `check_script` inside `check_if_startup_script_end` for easier debugging?

Reply:
That should be fine, and you could call a BashOperator to call a script. Something like below:

```
    provision = BashOperator(
        task_id='provision_' + task_id_suffix,
        bash_command=("bash your_check_script.sh"),
    )
```

You could find some examples [here](https://source.corp.google.com/piper///depot/google3/experimental/users/ranran/xlmlAirflow/demo_dags/utils/xlml_controller.py;l=60).

Reply:
Added all bash scripts into xlml/utils/startup_script_util.py. It should be cleaner now!